arangod/Cluster/ClusterIndexMethods.cpp

```diff
@@ -886,10 +886,21 @@ auto ensureIndexCoordinatorReplication2Inner(
                   }
                   return false;
                 })
-            .thenValue([&clusterInfo](auto index) {
-              // Need to wait here, until ClusterInfo has updated to latest
-              // plan.
-              return clusterInfo.waitForPlan(index);
+            .thenValue([&clusterInfo,
+                        &server](auto index) -> futures::Future<Result> {
+              if (clusterInfo.getPlanIndex() < index) {
+                // Need to wait here, until ClusterInfo has updated to latest
+                // plan.
+                auto& agencyCache =
+                    server.getFeature<ClusterFeature>().agencyCache();
+                auto [version, _] = agencyCache.read(std::vector<std::string>{
+                    AgencyCommHelper::path("Plan/Version")});
+                auto planVersion = version->slice()[0]
+                                       .get({"arango", "Plan", "Version"})
+                                       .getNumber<uint64_t>();
+                return clusterInfo.waitForPlanVersion(planVersion);
```

Comment:
Note the existence of [ClusterInfo::fetchAndWaitForPlanVersion() ](https://github.com/arangodb/arangodb/blob/devel/arangod/Cluster/ClusterInfo.cpp#L5587) which serves the same purpose in [ReplicatedLogMethodsCoordinator::createReplicatedLog()](https://github.com/arangodb/arangodb/blob/devel/arangod/Replication2/Methods.cpp#L423).
The only difference is that it sends a request to the agency, while we're using the agency cache here, and it also works based on a timeout. Happy to switch to that method if this patch seems redundant.

Reply:
Please keep this version, as we save the unnecessary call to the agency.
All required information is available locally.